### PR TITLE
TINKERPOP-1284: StarGraph does not handle self-loops correctly.

### PR DESCRIPTION
**Pull request by okram**
https://issues.apache.org/jira/browse/TINKERPOP-1284

```
gremlin> graph = TinkerGraph.open()
==>tinkergraph[vertices:0 edges:0]
gremlin>
gremlin> v = graph.addVertex()
==>v[0]
gremlin> v.addEdge("self",v)
==>e[1][0-self->0]
gremlin> g = graph.traversal()
==>graphtraversalsource[tinkergraph[vertices:1 edges:1], standard]
gremlin> g.V(0l)
==>v[0]
gremlin>  g.V(0l).bothE()
==>e[1][0-self->0]
==>e[1][0-self->0]
gremlin> g.V(0l).outE()
==>e[1][0-self->0]
gremlin> g.V(0l).inE()
==>e[1][0-self->0]
gremlin>
gremlin>
gremlin> starGraph = org.apache.tinkerpop.gremlin.structure.util.star.StarGraph.of(v)
==>stargraph[starOf:v[0]]
gremlin> sg = starGraph.traversal()
==>graphtraversalsource[stargraph[starOf:v[0]], standard]
gremlin> sg.V(0l)
==>v[0]
gremlin> sg.V(0l).bothE()
==>e[1][0-self->0]
==>e[1][0-self->0]
gremlin> sg.V(0l).outE()
==>e[1][0-self->0]
gremlin> sg.V(0l).inE()
==>e[1][0-self->0]
gremlin>
gremlin> // TADA!
==>true
gremlin>
```

`mvn clean install` and Spark integration tests.

VOTE +1

### CODE REVIEW

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/star/StarGraph.java [hunk 1]
```diff
@@ -290,7 +288,16 @@ public void dropVertexProperties(final String... propertyKeys) {
 
         @Override
         public Edge addEdge(final String label, final Vertex inVertex, final Object... keyValues) {
-            return this.addOutEdge(label, inVertex, keyValues);
+            final Edge edge = this.addOutEdge(label, inVertex, keyValues);
+            if (inVertex.equals(this)) {
+                List<Edge> inE = inEdges.get(label);
```
Comment (dalaro) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/star/StarGraph.java]: inEdges may be null here on a pristine StarVertex

**Comment by dalaro** [comment]
+1 (non-binding)

This change fixed my formerly-broken StarGraph self-edge use case.

I tested this PR's changes against 3.2 instead of 3.1.  Specifically, I checked out  e7b3267c99fe62cee0e887ca2b2d9aee17381631 and did `git cherry-pick -x apache/master..apache/TINKERPOP-1284`.

**Comment by dkuppitz** [comment]
`docker/build.sh -t -i -n` succeeded.

VOTE: +1

**Comment by spmallette** [comment]
This is a good fix. PR says it is targeted for master but it is really for tp31. 

VOTE +1

**Pull request merged**